File: nodejs/MinioNodejs.Services/src/interfaces/IMinioService.ts
import { Readable } from "stream";

export interface IMinioService {
  createBucket(bucketName: string): Promise<void>;
  listBuckets(): Promise<string[]>;
  bucketExists(bucketName: string): Promise<boolean>;
  deleteBucket(bucketName: string): Promise<void>;
  uploadFile(
    bucketName: string,
    objectName: string,
    stream: Readable,
    size: number,
    contentType: string
  ): Promise<void>;
  downloadFile(bucketName: string, objectName: string): Promise<Readable>;
}

File: java/MinioJava.Services/src/main/java/com/example/minio/services/IMinioService.java
package com.example.minio.services;

import java.io.InputStream;
import java.util.List;

public interface IMinioService {
    void createBucket(String bucketName) throws Exception;
    List<String> listBuckets() throws Exception;
    boolean bucketExists(String bucketName) throws Exception;
    void deleteBucket(String bucketName) throws Exception;
    void uploadFile(String bucketName, String objectName, InputStream stream, long size, String contentType) throws Exception;
    InputStream downloadFile(String bucketName, String objectName) throws Exception;
}
